docs/getting-started/index.mdx

```diff
@@ -6,174 +6,377 @@ pagination_next: http/index
 
+```bash
+ngrok http 8080 --url <YOUR_NGROK_DOMAIN>
 ```
-ngrok http 8080 --url jumpy-red-mollusk.ngrok-free.app
-```
 
-## Step 5: Secure your app
+## Step 4: Write your first Traffic Policy rules
+
+You may not want anyone to be able to access your app or API on your even though
+you've made it publicly available. With ngrok, you can add authentication and
+other traffic management rules, without making any changes to your service,
+using our [Traffic Policy](/docs/http/traffic-policy/index.mdx) engine.
+
+Traffic Policy is a powerful configuration language for your endpoints, letting
+you flexibly filter, match, manage, and orchestrate traffic exactly as you need.
 
-You may not want everyone to be able to access your application. ngrok can
-quickly add authentication to your app without any changes. If your Google
-account is `alan@example.com`, you can restrict access only for yourself by
-running ngrok with:
+Let's look at a basic Traffic Policy rule to illustate the key concepts:
 
+```yaml
+on_http_request:
+  - expressions:
+      - ""conn.ts.start < timestamp('2999-12-31T23:59:59Z')""
+    actions:
+      - type: custom-response
+        config:
+          content: "Hello, world!"
 ```
-ngrok http http://localhost:8080 --oauth=google --oauth-allow-email=alan@example.com
+
+<p className="text-xl">`on_http_request`</p>
+
+A [phase](/docs/traffic-policy/concepts/phases.mdx), which is a distinct point in the lifecycle of a request, where you can
+inspect and manage traffic.
+
+In this case, we're acting on the moment a request reaches the ngrok network.
+
+<p className="text-xl">`expressions`</p>
+
+One or more conditions, written in Common Expression Language (CEL) that must
+evaluate to `true` to trigger the corresponding action. You can use
+[variables](/docs/traffic-policy/variables/index.mdx) and
+[macros](/docs/traffic-policy/macros/index.mdx) to define your
+[expressions](/docs/traffic-policy/concepts/expressions.mdx).
+
+This particular expression checks whether the time a request reaches ngrok is
+before the moment we all roll into year 3000, so it'll evaluate as `true` for
+the time being.
+
+<p className="text-xl">`actions`</p>
+
+Operations that take specific
+[action](/docs/traffic-policy/concepts/actions.mdx) on your traffic during the
+corresponding phase of the request lifecycle, if all expressions evaluate as
+`true`. If there are no expressions, then Traffic Policy will always run the
+action(s).
+
+Here we use the [custom response
+action](/docs/http/traffic-policy/actions/custom-response.mdx) to return `Hello,
+world!`.
+
+### Secure your service with OAuth or basic auth
+
+Let's say you want to restrict access only to you and your coworkers by
+requiring a successful OAuth log-in with Google, denying anyone who tries to
+login using an email that doesn't belong to your company's domain name.
+
+Create your first Traffic Policy rule by creating a new file called `auth.yaml`
```

Comment:
do we need to use a policy file here? why not the command line flags? this is a quickstart after all.